include/eve/detail/function/canonical_shuffle_adapter.hpp

```diff
@@ -0,0 +1,59 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/pattern.hpp>
+
+namespace eve
+{
+  EVE_REGISTER_CALLABLE(canonical_shuffle_adapter_impl_)
+  EVE_DECLARE_CALLABLE(canonical_shuffle_adapter_impl_, canonical_shuffle_adapter_impl)
+
+  namespace detail
+  {
+    EVE_ALIAS_CALLABLE(canonical_shuffle_adapter_impl_, canonical_shuffle_adapter_impl);
+  } // namespace detail
+
+  EVE_CALLABLE_API(canonical_shuffle_adapter_impl_, canonical_shuffle_adapter_impl)
```

Comment:
should this be public ?

Reply:
I was thinking about that...
I think that if users gonna write custom shuffles, they will just call the intrinsic.

I can't imagine why they'd need this